src/Storages/MergeTree/MergedBlockOutputStream.cpp

```diff
@@ -169,6 +169,15 @@ MergedBlockOutputStream::Finalizer MergedBlockOutputStream::finalizePartAsync(
         new_part->setColumns(part_columns, serialization_infos, metadata_snapshot->getMetadataVersion());
     }
 
+    if (storage.partMetadataFormatVersion() >= PART_METADATA_FORMAT_VERSION_INITIAL)
+    {
+        Poco::JSON::Object::Ptr metajson = new Poco::JSON::Object();
+        metajson->set("version", storage.partMetadataFormatVersion().toUnderType());
+        unsigned long t = time(nullptr);
```

Comment:
Just occurred to me that this will also not be consistent between machines, so checksum mismatches would occur anyways. I'll try to change it to use something consistent.

Reply:
In most of the replication-related tasks the time of the replication queue entry is used to avoid that.